ui/src/store.js

```diff
@@ -0,0 +1,186 @@
+import autodux from 'autodux';
+import { doFetch } from './utils/fetch-websocket';
+
+export const {
+  reducer,
+  initial: defaultState,
+  actions: {
+    activateConnection,
+    deactivateConnection,
+    serverConnected,
+    serverDisconnected,
+    updatePurses,
+    changePurse,
+    swapInputs,
+    changeAmount,
+    createOffer,
+    resetState,
+    updateInvitationDepositId,
+  },
+} = autodux({
+  slice: 'autoswap',
+  initial: {
+    active: false,
+    connected: false,
+    account: null,
+    purses: null,
+    inputPurse: null,
+    outputPurse: null,
+    inputAmount: null,
+    outputAmount: null,
+    freeVariable: null,
+    invitationDepositId: null,
+  },
+  actions: {
+    activateConnection: state => ({ ...state, active: true }),
+    deactivateConnection: state => ({ ...state, active: false }),
+    serverConnected: state => ({ ...state, connected: true }),
+    serverDisconnected: state => ({ ...state, connected: false }),
+    updatePurses: (state, purses) => ({ ...state, purses }),
+    changePurse: {
+      // Map positional arguments.
+      create: (purse, fieldNumber, freeVariable = null) => ({
```

Comment:
this `create`/`reducer` seems like a pattern. Is it a pattern of ours?  A thing from autodux?  How does `reducer` get invoked?

Reply:
It's an autodux thing.  `create` takes the action's arguments as input and creates a second argument to `reducer`.